src/modules/settings/MultiSetting.ts

```diff
@@ -0,0 +1,30 @@
+import BaseSetting from './BaseSetting';
+
+export default class MultiSetting<T> extends BaseSetting<T[], T> {
+    private optionValues;
+
+    validValue(value: T[]): boolean {
+        if (!this.isUnlocked(value)) {
+            return false;
+        }
+
+        if (value.length === 0) {
+            return true;
+        }
+        if (this.options.length === 0) {
+            return false;
+        }
+
+        if (this.optionValues === undefined) {
+            this.optionValues = this.options.map((o) => o.value);
```

Comment:
Would be cleaner to create this in a constructor, and probably better to use a Set since we are just using to check if values exists in there.

Reply:
I tried doing it in the Constructor at first but I kept having errors where it wasn't being set and I have no idea why. This was the only thing that seemed to work

Reply:
The options are passed into the constructor so I don't see any reason it wouldn't be able to set this there?

Can you share how you were doing it?

Reply:
The base constructor calls validValue on the default value, but we can't set the optionValues before this as we are not allowed to access 'this' before calling 'super'
The only alternatives I can think of are to remove the call to validValue for the default value or to create a new empty function called in the base constructor which sets the optionValues in an override

Reply:
Guess we can just leave it for now then